internal/common/platforms.go

```diff
@@ -0,0 +1,45 @@
+package common
+
+import (
+	"github.com/openshift/assisted-service/models"
+)
+
+const (
+	VmwareManufacturer string = "VMware, Inc."
+)
+
+var supportedManufacturers = map[string]models.PlatformType{
+	VmwareManufacturer: models.PlatformTypeVsphere,
+}
+
+func GetPlatformFromVendor(vendor models.SystemVendor) models.PlatformType {
+	return supportedManufacturers[vendor.Manufacturer]
+}
+
+func GetHostSupportedPlatforms(inventory models.Inventory) *[]models.PlatformType {
+	hostPlatform := []models.PlatformType{models.PlatformTypeBaremetal}
+
+	if platform := GetPlatformFromVendor(*inventory.SystemVendor); platform != "" {
+		hostPlatform = append(hostPlatform, platform)
+	}
+
+	return &hostPlatform
+}
+
+func GetClusterSupportedPlatforms(cluster Cluster, hostsPlatformsCounter map[models.PlatformType]int) *[]models.PlatformType {
```

Comment:
Let's try to do it more clear and simple using intersect instead of using a counter and maybe get the host list here